tools/flaky_tests/Jenkinsfile

```diff
@@ -0,0 +1,58 @@
+// -*- mode: groovy -*-
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+err = null
+
+def init_git() {
+    deleteDir()
+    retry(5) {
+        try{
+            timeout(time: 15, unit: 'MINUTES') {
+                checkout scm
+                sh 'git submodule update --init --recursive'
+                sh 'git clean -d -f'
+            }
+        } catch (exc) {
+            deleteDir()
+            error "Failed to fetch source code with $(exc)"
+            sleep 2
+        }
+    }
+}
+
+def docker_run(platform, function_name, use_nvidia, shared_mem = '500m') {
+    def command = "ci/build.py --docker-registry ${env.DOCKER_CACHE_REGISTRY} %USE_NVIDIA% --platform %PLATFORM% --docker-build-retries 3 --shm-size  %%SHARED_MEM% /work/runtime_functions.sh %FUNCTION_NAME%"
+    command = command.replaceAll('%USE_NVIDIA%', use_nvidia ? '--nvidiadocker': '')
+    command = command.replaceAll('%PLATFORM%', platform)
+    command = command.replaceAll('%FUNCTION_NAME%', function_name)
+    command = command.replaceAll('%SHARED_MEM%', shared_mem)
+
+    sh command
+}
+
+try{
+    stage('Flakiness Check'){
+        node('mxnetlinux-gpu') {
+            ws('workspace/flakiness_check'){
+                init_git()
+                docker_run('ubuntu_gpu', 'run_flakiness_checker', true)
```

Comment:
Does it not need a compiled version of MXNet?

Reply:
right, for now I'll build mxnet each time we run this, but really we should only do so when dependency analyzer detects changed tests. This would take some refactoring, however, and I think it would be good to get a version of this tool running ASAP.